pandas/tests/dtypes/cast/test_downcast.py

```diff
@@ -34,6 +37,13 @@
             "int64",
             np.array([decimal.Decimal(0.0)]),
         ),
+        (
+            # GH#45837
+            np.array([Timedelta(days=1), Timedelta(days=2)], dtype=object),
+            "infer",
+            np.array([1, 2], dtype="m8[D]").astype("m8[ns]"),
+        ),
+        # TODO: similar for dt64, dt64tz, Period, Interval?
```

Comment:
xfail these cases?

Reply:
not yet sure what the correct behavior is.  will make xfails once confident